modules/auxiliary/admin/mssql/mssql_escalate_execute_as.rb

```diff
@@ -17,14 +18,19 @@ def initialize(info = {})
       },
       'Author'      => ['nullbind <scott.sutherland[at]netspi.com>'],
       'License'     => MSF_LICENSE,
-      'References'  => [['URL','http://msdn.microsoft.com/en-us/library/ms178640.aspx']]
+      'References'  => [['URL','http://msdn.microsoft.com/en-us/library/ms178640.aspx']],
+      'SessionTypes' => %w[MSSQL]
     ))
   end
 
   def run
+    if (datastore['SESSION'] && session)
+      set_session(session)
+    end
     # Check connection and issue initial query
-    print_status("Attempting to connect to the database server at #{rhost}:#{rport} as #{datastore['USERNAME']}...")
-    if mssql_login_datastore
+    print_status("Attempting to connect to the database server at #{datastore['RHOST']}:#{datastore['RPORT']} as #{datastore['USERNAME']}...")
+
+    if (datastore['SESSION'] && session) || mssql_login_datastore
```

Comment:
I don't think you need to be duplicating these checks for a session I think something like 
```
if session
  set_session(session)
else
   mssql_login_datastore
 end
 ```
 works for what you want?